docs/client/cli/kyuubi_beeline.md

```diff
@@ -0,0 +1,77 @@
+- The ASF licenses this file to You under the Apache License, Version 2.0
+- (the "License"); you may not use this file except in compliance with
+- the License.  You may obtain a copy of the License at
+-
+-   http://www.apache.org/licenses/LICENSE-2.0
+-
+- Unless required by applicable law or agreed to in writing, software
+- distributed under the License is distributed on an "AS IS" BASIS,
+- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+- See the License for the specific language governing permissions and
+- limitations under the License.
+-->
+
+# Kyuubi Beeline
+
+## What is Kyuubi Beeline
+
+Kyuubi Beeline is a Command Line Shell that uses JDBC driver to connect to Kyuubi server, HiveServer2 and other
+HiveServer2 compatible servers, like Impala, Flink SQL Gateway, to execute queries. Kyuubi Beeline is derived from
+[Hive Beeline](https://cwiki.apache.org/confluence/display/Hive/HiveServer2+Clients#HiveServer2Clients-Beeline%E2%80%93CommandLineShell),
+almost functionalities of Hive Beeline should be applicable to Kyuubi Beeline as well.
+
+Note: Kyuubi Beeline removes the support of "embedded mode" because it is coupled with Apache Hive implementation details,
+thus only "remote mode" can be used in Kyuubi Beeline. 
+
+##  Kyuubi Beeline properties files
+
+Kyuubi Beeline automatically constructs a connection URL based on some configuration properties in below four configuration files.
+
+- `beeline-site.xml`: By configuring named URLs within the `beeline-site.xml` file, users can employ the command `beeline -c <named_url>` to connect to predefined Kyuubi servers.
+
+- `beeline-hs2-connection.xml`: Users are able to set various properties in this file to change Kyuubi Beeline's behavior when connecting to Kyuubi servers. It provides user-specific connection properties, such as username, password, and other connection properties, such as SSL configuration properties, etc.
+
+- `kyuubi-defaults.conf`: The main configuration file for Apache Kyuubi. It contains global configuration properties in Kyuubi server like hostname and port information, Kerberos configuration properties.
+
+- `hive-site.xml`: It is one of the main configuration files for Apache Hive. It contains global configuration properties in Hive services like HS2 hostname and port information, Kerberos configuration properties, SSL properties, etc.
+
+## How users automatically connect to Kyuubi Servers via properties files
+
+Kyuubi Beeline looks for all configuration files sequentially once it gets started.
+
+1. `beeline-site.xml`: Kyuubi Beeline starts by retrieving the named URL from this file.
+
+2. `beeline-hs2-connection.xml`: This file is checked next, and any properties in this file take precedence over those in `beeline-site.xml`.
+
+3. `kyuubi-defaults.conf`: Properties from this file are loaded next. Values in this file may merge with the values of hiveconf from `beeline-hs2-connection.xml`.
```

Comment:
the "merge" behavior is confusing here, which one will take effect for duplicated properties?

Reply:
They are both appended to the URL, but the latter one will take effect in case of duplicated properties.